### PR TITLE
fix(protocol-designer) fix the slot details glitch when wrapping deck view

### PR DESCRIPTION
**Pull request by koji**
# Overview
- fix the slot details glitch when wrapping deck view by adding debounce 
- create a new component for Protocol Starting Deck part add its test
- fix Liquid Details condition to display N/A 

close RQA-3353

## Test Plan and Hands on Testing
1. create or import a protocol
2. change windows size 1024px
3. hovering slot
check that glitching doesn't happen any more

## Review requests
low

## Risk assessment
low

### CODE REVIEW

File: protocol-designer/src/pages/ProtocolOverview/__tests__/ProtocolOverview.test.tsx [hunk 1]
```diff
@@ -80,10 +78,10 @@ describe('ProtocolOverview', () => {
...
       <div>mock MaterialsListModal</div>
     )
-    vi.mocked(DeckThumbnail).mockReturnValue(<div>mock DeckThumbnail</div>)
-    vi.mocked(OffDeckThumbnail).mockReturnValue(
-      <div>mock OffdeckThumbnail</div>
-    )
+    // vi.mocked(DeckThumbnail).mockReturnValue(<div>mock DeckThumbnail</div>)
+    // vi.mocked(OffDeckThumbnail).mockReturnValue(
+    //   <div>mock OffdeckThumbnail</div>
+    // )
```
Comment (jerader) [inline protocol-designer/src/pages/ProtocolOverview/__tests__/ProtocolOverview.test.tsx]: i guess these can be deleted now?

**Review by jerader — COMMENTED** [review]
The debounce i believe makes it take a second for the slot details to render, is that expected? 
https://github.com/user-attachments/assets/47267cda-adf7-4b34-a529-ffeece4d36eb

**Comment by koji** [comment]
> The debounce i believe makes it take a second for the slot details to render, is that expected? https://github.com/user-attachments/assets/47267cda-adf7-4b34-a529-ffeece4d36eb

yes. the issue is that rendering is too fast between slots.

**Comment by jerader** [comment]
> > The debounce i believe makes it take a second for the slot details to render, is that expected? https://github.com/user-attachments/assets/47267cda-adf7-4b34-a529-ffeece4d36eb
> 
> yes. the issue is that rendering is too fast between slots.

cool, makes sense to me. Can you double check with design that this behavior is ok? i don't think there is much we can do code-wise though to fix that.

**Review by jerader — APPROVED** [review]
seems like an okay fix. If over the next few weeks design/qa/etc don't like the debounce, we can possibly revisit for other options.

**Pull request merged**